src/types/app_config.go

```diff
@@ -125,3 +126,18 @@ func (a AppConfig) forEachService(fn func(string, string, ServiceData)) {
 		fn("public", serviceName, data)
 	}
 }
+
+func (a AppConfig) validateAppConfig() error {
+	appNameRegex := regexp.MustCompile("^[a-z0-9]+(-[a-z0-9]+)*$")
+	if !appNameRegex.MatchString(a.Name) {
+		return errors.New("only lowercase alphanumeric character(s) separated by a single hyphen are allowed. Must match regex: /^[a-z0-9]+(-[a-z0-9]+)*$/")
+	}
+	var err error
+	serviceRoleRegex := regexp.MustCompile("^[a-zA-Z0-9]+(-[a-zA-Z0-9]+)*$")
+	a.forEachService(func(serviceType, serviceRole string, data ServiceData) {
+		if !serviceRoleRegex.MatchString(serviceRole) {
+			err = errors.New("only alphanumeric character(s) separated by a single hyphen are allowed. Must match regex: /^[a-zA-Z0-9]+(-[a-zA-Z0-9]+)*$/")
```

Comment:
Can this error message include that it is in `application.yml` and that its under `services.{serviceType}` and the offending key?

Maybe something like `The "services.public" key "invalid-service!" in application.yml in invalid. ...`